src/gc/unix/cgroup.cpp

```diff
@@ -0,0 +1,271 @@
+                goto done;
+            }
+    
+            if (strncmp(filesystemType, "cgroup", 6) == 0)
+            {
+                char* context = NULL;
+                char* strTok = strtok_r(options, ",", &context); 
+                while (strTok != NULL)
+                {
+                    if (strncmp("memory", strTok, 6) == 0)
+                    {
+                        sscanfRet = sscanf(line,
+                                           "%*d %*d %*s %*s %s ",
+                                           mountpath);
+                        if (sscanfRet != 1)
+                        {
+                            assert(!"Failed to parse mount info file contents with sscanf.");
+                            goto done;
+                        }
+                        retval = true;
+                        break;
+                    }
+                    strTok = strtok_r(NULL, ",", &context);
+                }
+            }
+        }
+    done:
+        if (line)
+            free(line);
+        if (mountinfofile)
+            fclose(mountinfofile);
+        return retval;
+    }
+    
+    bool FindCGroupPathForMemorySubsystem(char* cgrouppath)
+    {
+        bool retval = false;
+        char *line = nullptr;
+        size_t lineLen = 0;
+        FILE *cgroupfile = fopen(PROC_CGROUP_FILENAME, "r");
+        if (cgroupfile == NULL)
+            goto done;
+    
+        while (!retval && getline(&line, &lineLen, cgroupfile) != -1)
+        {
+            char subsystem_list[100];
+            char cgroup_path[MAX_PATH];
+    
+            // See man page of proc to get format for /proc/self/cgroup file
+            int sscanfRet = sscanf(line, 
+                                   "%*[^:]:%[^:]:%s",
+                                   subsystem_list,
+                                   cgroup_path);
+            if (sscanfRet != 2)
+            {
+                assert(!"Failed to parse cgroup info file contents with sscanf.");
+                goto done;
+            }
+    
+            char* context = NULL;
+            char* strTok = strtok_r(subsystem_list, ",", &context); 
+            while (strTok != NULL)
+            {
+                if (strncmp("memory", strTok, 6) == 0)
+                {
+                    strcpy(cgrouppath, cgroup_path);
+                    retval = true;
+                    break;  
+                }
+                strTok = strtok_r(NULL, ",", &context);
+            }
+        }
+    done:
+        if (line)
+            free(line);
+        if (cgroupfile)
+            fclose(cgroupfile);
+        return retval;
+    }
+    
+    bool ReadMemoryValueFromFile(const char* filename, size_t* val)
+    {
+        bool result = false;
+        char *line = nullptr;
+        size_t lineLen = 0;
+        char* endptr = NULL;
+        size_t num = 0, l, multiplier;
+        FILE* file = NULL;
+    
+        if (val == NULL)
```

Comment:
A nit - you have this check only here, but not in the other copy of the function.